storage/innobase/sync/sync0rw.cc

```diff
@@ -307,7 +310,7 @@ rw_lock_s_lock_spin(
 		os_thread_yield();
 	}
 
-	++spin_count;
+	spin_count += lint(i - j);
```

Comment:
No `j` please. `spin_count += i` is almost enough. There's just one missing `i = 0` reset, which doesn't seem to make any sense.

All in all I'd rather suggest to move `ulint i = 0` declaration under `lock_loop:` label and remove `i = 0` around the end of this function.

Reply:
That's doesn't serve the purpose. Let me put the idea so we are on same page.
Logic is supposed to perform N rounds (where N = 30). The loop can get interrupted say when i = 18 then spin_count = 18 but this thread could still fail to obtain the lock because some other thread got it (in meantime) so it will go back to completes it 30 rounds wait. Next time it is interrupted say when i = 24 then we just need to add 6 to spin_count so spin_count = 24. Resetting i = 0 would mean everytime a loop is interrupted it would restart with fresh 30 rounds. Idea is to restart the loop from the point it has left.

So when is i reset to 0. i is resetted to 0 when spin-loop fails to obtain lock and flow forces to enter os-wait.

<snippet>
                i = 0;

                goto lock_loop;
</snippet>